compiler/noirc_evaluator/src/ssa/acir_gen/mod.rs

```diff
@@ -821,11 +858,29 @@ impl Context {
         mut var_index: AcirVar,
         dfg: &DataFlowGraph,
     ) -> Result<AcirValue, RuntimeError> {
-        let (_, _, block_id) = self.check_array_is_initialized(array, dfg)?;
+        let (array_id, _, block_id) = self.check_array_is_initialized(array, dfg)?;
 
         let results = dfg.instruction_results(instruction);
         let res_typ = dfg.type_of_value(results[0]);
-        let value = self.array_get_value(&res_typ, block_id, &mut var_index)?;
+
+        let value = if !res_typ.contains_slice_element() {
+            self.array_get_value(&res_typ, block_id, &mut var_index, &[])?
+        } else {
+            let mut slice_sizes = self
+                .slice_sizes
+                .get(&array_id)
+                .expect("ICE: Array with slices should have associated slice sizes")
+                .clone();
+
+            slice_sizes.remove(0);
```

Comment:
```suggestion
            let slice_sizes = self
                .slice_sizes
                .get(&array_id)
                .expect("ICE: Array with slices should have associated slice sizes");

            let slice_sizes = slice_sizes[1..].to_vec();
```

Nit but we can avoid the shifting from `.remove` here.

Reply:
Also I forget, we're slicing off the first element as it's holding the length of the array, correct? Could we sprinkle some comments in places where we're adding/removing this element and it's not obvious from how we're using it?

Reply:
I'm not sure what the time difference is with a clone vs. a shift but I accepted the suggestion.

Reply:
I added a couple comments for this as well.

Reply:
> I'm not sure what the time difference is with a clone vs. a shift but I accepted the suggestion.

In the first case we're cloning then removing, both O(n).
In the second case we're creating a slice from the array O(1), then cloning O(n). So slicing first instead of removing after removes a O(n) operation.